### DISCUSSION TITLE
[GUIDANCE] [SP-API] Vendor Inventory data

### DISCUSSION THREAD

**Discussion by dannybpw** [amzn/selling-partner-api-models#3078] [General]
Hello - looking for some guidance regarding the usage of the SPAPI for Vendors.

We are looking for a way to retrieve current inventory for vendors but unable to locate any endpoints that may achieve this.

The closest we came was the Reports API with GET_VENDOR_INVENTORY_REPORT, but this is only available to private apps.

Is there any way to retrieve inventory information using the API for Vendors as a public app?

Thank you in advance.

**Comment by MeenaAmz** [comment]
GET_VENDOR_INVENTORY_REPORT is available to vendors who have the Brand Analytics Selling Partner API role. Does your app have this role and you still cant access this report for a vendor who authorized your app?

**Comment by sungolivia** [comment]
@dannybpw have you had a chance to check out the answer above?

**Comment by dannybpw** [comment]
Sorry for the delay on this, and appreciate the replies!

@MeenaAmz @sungolivia - the GET_VENDOR_INVENTORY_REPORT is only available to vendors who have the Brand Analytics Selling Partner API role as per the SPAPI documentation: https://developer-docs.amazon.com/sp-api/docs/report-type-values

We were told that this role is only for private apps, but we are developing a public app.

Are you able to advise on whether there is any way to retrieve inventory updates for vendors as a **public app**?

Thank you

**Comment by sungolivia** [comment]
hey @dannybpw can you tell me who told you that role is only for private apps?
It is available to public devs as well!

Make sure you check this out if you don't see the option for the role in your application configurations!
https://developer-docs.amazon.com/sp-api-blog/docs/choosing-roles-for-developer-profile-and-sp-api-app

**Reply by dannybpw** [reply]
The seller [central](https://sellercentral.amazon.ca/) support told me that in a ticket that I opened.

```
I understand that you are requesting GET_VENDOR_INVENTORY_REPORT.

Even though you completed the developer registration, the level of access you were given on the registration does not allow you to get Brand Reports.

Developers can view their current SP-API role assignments or make updates at any time within their Developer Profile:
https://sellercentral.amazon.com/developer/register

Please request access to Brand Analytics to have access to this report.

Note: Only Private Developers that are Brand Owners can access this feature.
```

We are currently in the process of updating our listing so will see if those roles are available to us once we are approved so will see if it is available.
